src/mgos_config_util.c

```diff
@@ -176,12 +197,21 @@ void mgos_conf_parse_cb(void *data, const char *name, size_t name_len,
       if (tok->len > 0) {
         s = (char *) malloc(tok->len + 1);
         if (s == NULL) {
+          mg_asprintf(msg, 0, "insufficient memory");
+          if (ctx->msg) {
+            *ctx->msg = *msg;
+          }
           ctx->result = false;
           return;
         }
         int n = json_unescape(tok->ptr, tok->len, s, tok->len);
         if (n < 0) {
           free(s);
+          mg_asprintf(msg, 0, "[%s] invalid string", path);
+          LOG(LL_ERROR, (*msg));
+          if (ctx->msg) {
+            *ctx->msg = *msg;
```

Comment:
we need to make sure *msg doesn't leak.

Reply:
A note saying it should be freed by the caller?

Reply:
yes, that'd be nice (in the header).
but when it's not provided, we should still not be leaking.